### DISCUSSION TITLE
New user help: Create and Start instance help

### DISCUSSION THREAD

**Discussion by diegobkc** [flowsforapex/apex-flowsforapex#778] [Q&A] [answered]
Hello, I am following along with one of the introduction tutorials and while trying to create and start a workflow process, I get the error: ORA-06550: line2, column 50: PLS-00201: identifier 'Flow_plugin_manage_instance.execution' must be declared.  I have followed all of the steps listed but I cannot find any reason why I would be generating this error.  If anyone has seen this before I could use some assistance. 
![Screenshot 2024-12-24 062517](https://github.com/user-attachments/assets/4de3f476-7db4-4774-b583-8a3a1d4f95f4)
![Screenshot 2024-12-24 062606](https://github.com/user-attachments/assets/005a1920-5ba8-46b2-a1d8-b071cd5ac876)

**Comment by rallen2010** [comment]
Sorry that you're having problems - I'll look into this and get back to you in a few days (after Christmas)!

**Reply by diegobkc** [reply]
Thank You. I also copied this post into Issues. I was not sure of the correct location for it.  Thank you for your assistance.

**Reply by diegobkc** [reply]
is_internal_error: false
original_additional_info: ORA-06550: line 2, column 50: PLS-00201: identifier 'FLOW_PLUGIN_MANAGE_INSTANCE.EXECUTION' must be declared
ora_sqlcode: -6550
ora_sqlerrm: ORA-06550: line 2, column 50: PLS-00201: identifier 'FLOW_PLUGIN_MANAGE_INSTANCE.EXECUTION' must be declared ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 155 ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2676 ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_240100", line 732 ORA-06550: line 2, column 7: PL/SQL: Statement ignored ORA-06512: at "SYS.DBMS_SYS_SQL", line 1458 ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_240100", line 720 ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2015 ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2593 ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 130 ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 436 ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC", line 553 ORA-06512: at "APEX_240100.WWV_FLOW_PLUGIN", line 3173 ORA-06512: at "APEX_240100.WWV_FLOW_PLUGIN", line 4293
component.type: APEX_APPLICATION_PAGE_PROCESS
component.id: 43125724092248814
component.name: Start Workflow
error_backtrace:
ORA-06512: at "SYS.DBMS_SYS_SQL", line 1458
ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_240100", line 720
ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 155
ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2676
ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_240100", line 732
ORA-06512: at "SYS.DBMS_SYS_SQL", line 1458
ORA-06512: at "SYS.WWV_DBMS_SQL_APEX_240100", line 720
ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2015
ORA-06512: at "APEX_240100.WWV_FLOW_DYNAMIC_EXEC", line 2593
ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 130
ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC_PLSQL", line 436
ORA-06512: at "APEX_240100.WWV_FLOW_CODE_EXEC", line 553
ORA-06512: at "APEX_240100.WWV_FLOW_PLUGIN", line 3173
ORA-06512: at "APEX_240100.WWV_FLOW_PLUGIN", line 4293
ORA-06512: at "APEX_240100.WWV_FLOW_PROCESS", line 192
error_statement:
begin declare
begin wwv_flow_plugin_api.g_process_exec_result:=flow_plugin_manage_instance.execution(p_process => wwv_flow_plugin_api.g_process,p_plugin => wwv_flow_plugin_api.g_plugin);end;
end;

**Reply by diegobkc** [reply]
![Screenshot 2024-12-27 231423](https://github.com/user-attachments/assets/1a507deb-d427-4d1f-97eb-27deec6bef78)
![Screenshot 2024-12-27 231403](https://github.com/user-attachments/assets/ae13b205-f5ac-4264-9380-0cec62cb79da)
![Screenshot 2024-12-27 231335](https://github.com/user-attachments/assets/4f82f3b8-afbb-4151-909b-6fc870db7f93)

**Comment by rallen2010** [comment]
HI - sorry for the delay in responding over Christmas and New Year.  I've just worked through the tutorial myself on a clean Flows for APEX 24.1 installation, following the instructions, and I can't replicate your problem.
The one thing that I noticed (and will change. today) is that under the second heading on the Getting Started page, the link from 'Installation of Flows for APEX' ....download the latest version of the Flows for APEX packaged app here...'  is pointing to our previous release, v23.1.   instead of v24.1 (should be https://github.com/flowsforapex/apex-flowsforapex/releases/download/v24.1/FlowsForAPEX_v24.1.zip).   I can't currently see any obvious reason for the error you got - but could you check the version of Flows for APEX you are using and the version of the plugins (see  app builder > shared components > plugins ) and check that everything is as expected.  If you are still having difficulty, please email me at info at Flowquest.net and we can set up a Zoom call to diagnose further....

**Reply by diegobkc** [reply]
The issue seems to be related to accessing the Flows for APEX application from a different schema than the one it was created in.  If I create another instance of the application for each schema, I don't have any issues at all. Should I be able to start a process that exist in another schema?

**Reply by Muddasar** [reply]
Hello dear @diegobkc 

Did you find the solution for this? I am facing same issue while creating instance from a different schema than the parsing schema of FFA application.

Please advise.

Regards,
Muddasar

**Reply by Muddasar** [reply]
Update:

I have created the public synonyms for FFA objects and granted the permissions to public as a workaround. Please correct me if I am doing anything wrong.

Find the attachment to get the script.
[FFA Public Synonyms and Grants.txt](https://github.com/user-attachments/files/20860870/FFA.Public.Synonyms.and.Grants.txt)

**Reply by rallen2010** [reply, marked as answer]
Flows for APEX is currently designed to install and operate in a single workspace and in the workspace schema.  In some future release, we might re-engineer the engine to make it a central installation - but this is a big task that isn't currently on our work list.  Running across schemas creates many issues, including:

 - ensuring that we don't break the isolation between APEX workspaces, thus breeching the APEX security policy
 - use of APEX Human Tasks and Task Lists in a workflow - which are also designed to work only in a single APEX Workspace.
 - ensuring our use of APEX meta data in, for example, the BPMN Modeler, can reference the correct objects
 - identifying the correct objects / namespaces when running background tasks, timer jobs, and (soon) asynchronous tasks
 - allowing independant upgrade of Flows for APEX for different workspaces on shared servers
 
Your solution of creating public synonyms and grants to public would have problems on shared servers - I think that if you take this approach, you should create private synonyms in your second schema and private grants for the Flows objects from the installed schema to the second schema to avoid any conflict with other installations. 
 
In the past, we have experimented with similar approaches, but found it difficult to resolve all of the issues to the extent that we are comfortable that this should be used for production use ... so we don't want to ship scripts to do this or to support it with the current release (24.1) or the upcoming release (25.1).
